src/lib.rs

```diff
@@ -1697,6 +1752,34 @@ mod tests {
         let (text, options) = unfill("foo\nbar");
         assert_eq!(text, "foo bar");
         assert_eq!(options.width, 3);
+        assert_eq!(options.line_ending, LineEnding::LF);
+    }
+
+    #[test]
+    fn unfill_no_new_line() {
+        let (text, options) = unfill("foo bar");
+        assert_eq!(text, "foo bar");
+        assert_eq!(options.width, 7);
+        assert_eq!(options.line_ending, LineEnding::LF);
+    }
+
+    #[test]
+    fn unfill_simple_crlf() {
```

Comment:
All tests where CRLF is invloved fail during the CI. Obviously, I don't see any issues locally.

From the assertion error it seems that `\r` is not removed. I will try to reproduce this behavior locally using the same container image as we use in the CI pipeline.

Reply:
> All tests where CRLF is invloved fail during the CI. Obviously, I don't see any issues locally.

Yeah, I could also not reproduce it at first until I tried with `cargo test --no-default-features`.

The problem is differing behavior of `display_width` depending on the enabled features. If the `unicode-width` feature is disabled, `display_width` will naively count each `char` as being 1 column wide. So we get

```
% cargo test --no-default-features tests::unfill_simple_crlf
  ...
test tests::unfill_simple_crlf ... FAILED

failures:

---- tests::unfill_simple_crlf stdout ----
[src/core.rs:185] &text = "foo\r"
[src/core.rs:185] width = 4
[src/core.rs:185] &text = "bar"
[src/core.rs:185] width = 3
thread 'tests::unfill_simple_crlf' panicked at 'assertion failed: `(left == right)`
  left: `4`,
 right: `3`', src/lib.rs:1771:9
```

The input string is `"foo\r\nbar"` and because we split on `\n` when counting the line width, we end up with the `"foo\r"` line.

Reply:
Thanks for your help.

As a simple and straightforward fix I decided to try `lines()` instead of `split('\n')` and to my surprise (I assumed `split` is there for a reason) I got better performance in the `unfill` benchmark (~19% difference in the same fashion I've got after implementing `NonEmptyLines` iterator). I fixed my cpu frequency to avoid noise and checked multiple times. The results are consistent.

Reply:
Great! I think I used `\n` because I was afraid to eat `\r` characters by accident. But this is now handled with the auto-detection.